sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -161,16 +161,49 @@ public HttpTransport(
    */
   @Override
   public boolean isRetryAfter(final @NotNull String type) {
-    if (sentryRetryAfterLimit.containsKey(type)) {
-      final Date date = sentryRetryAfterLimit.get(type);
+    final String category = getCategoryFromType(type);
+    final Date currentDate = new Date();
```

Comment:
yes, agreed, in this case, it's a single comparison and used only once, not sure if it'd pay off.
do we have this in cocoa or its just a room you saw for improvement?

Reply:
We created [`CurrentDateProvider`](https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentryCurrentDate.m), which is static. Be aware that static has the downside that if you run your tests in parallel it might get nasty. So we might change it again and pass a `CurrentDateProvider` as a dependency or pass the date as a method parameter would also work. For this code here it is not a big deal if we send a few events 2 or 3 milliseconds earlier. For other code you might want to make sure, that you don't have a off-by-one error for milliseconds or seconds and that is impossible to test with `Thread.sleep()`.

Reply:
yeah I agree, this should come from a ctor with package-private access modifier.
people implementing their own Transport should not know about it IMO.
I can improve that on another PR along with another `Date`s across the SDK.